braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java

```diff
@@ -205,172 +221,179 @@ public void onActivityDestroyed(@NonNull Activity activity) {
+        if (!TextUtils.isEmpty(lastName)) {
+            braze.getCurrentUser().setLastName(lastName);
+        }
 
-                    for (String key : traitsMap.keySet()) {
-                        if (RESERVED_KEY_SET.contains(key)) {
-                            continue;
-                        }
-                        Object value = traitsMap.get(key);
-                        if (value instanceof Boolean) {
-                            braze.getCurrentUser().setCustomUserAttribute(key, (Boolean) value);
-                        } else if (value instanceof Integer) {
-                            braze.getCurrentUser().setCustomUserAttribute(key, (Integer) value);
-                        } else if (value instanceof Double) {
-                            braze.getCurrentUser().setCustomUserAttribute(key, (Double) value);
-                        } else if (value instanceof Float) {
-                            braze.getCurrentUser().setCustomUserAttribute(key, (Float) value);
-                        } else if (value instanceof Long) {
-                            braze.getCurrentUser().setCustomUserAttribute(key, (Long) value);
-                        } else if (value instanceof Date) {
-                            long secondsFromEpoch = ((Date) value).getTime() / 1000L;
-                            braze.getCurrentUser().setCustomUserAttributeToSecondsFromEpoch(key, secondsFromEpoch);
-                        } else if (value instanceof String) {
-                            braze.getCurrentUser().setCustomUserAttribute(key, (String) value);
-                        } else {
-                            RudderLogger.logDebug("Braze can't map rudder value for custom Braze user "
-                                    + "attribute with key " + key + "and value " + value);
-                        }
-                    }
-                    break;
-                case MessageType.SCREEN:
-                    RudderLogger.logWarn("BrazeIntegrationFactory: MessageType is not supported");
-                    break;
-                default:
-                    RudderLogger.logWarn("BrazeIntegrationFactory: MessageType is not specified");
-                    break;
+        // gender
+        String gender = (String) this.needUpdate("gender", element);
+        if (!TextUtils.isEmpty(gender)) {
+            if (MALE_KEYS.contains(gender.toUpperCase())) {
+                braze.getCurrentUser().setGender(Gender.MALE);
+            } else if (FEMALE_KEYS.contains(gender.toUpperCase())) {
+                braze.getCurrentUser().setGender(Gender.FEMALE);
+            }
+        }
+
+        // phone
+        String phone = (String) this.needUpdate("phone", element);
+        if (!TextUtils.isEmpty(phone)) {
+            braze.getCurrentUser().setPhoneNumber(phone);
+        }
+
+        // address
+        RudderTraits.Address address = (RudderTraits.Address) this.needUpdate("address", element);
+        if (address != null) {
```

Comment:
Don't we set state, street, postalcode ?
If not required, the corresponding comparison on these 3 fields can be removed in `compareAddress`

Reply:
Addressed